### ISSUE TITLE
[FEAT] dm-verity for /usr

### ISSUE THREAD

**Issue by WavyEbuilder** [secureblue/secureblue#433]
To quote the kernel documentation:
"Device-Mapper’s “verity” target provides transparent integrity checking of block devices using a cryptographic digest provided by the kernel crypto API. This target is read-only."

Android makes use of this in its implementation of measured boot:
https://source.android.com/docs/security/features/verifiedboot/dm-
verity

Integration with ostree will be quite interesting to work on, considering it's not exactly an easy task from what I've seen, but maybe a long-term goal worth looking into.

- [x] I declare that I have read the README and my feature request is in-scope.

**Comment by TheHooly** [comment]
dm-verity is not compatible with rpm-ostree's layering, and would stop users from layering packages that they may need but are not part of the base image, e.g. Wifi drivers, virt-manager.

There is an effort to instead add composefs support to Fedora Atomic upstream, which would add fs-verity as an integrity seal instead, without precluding layering via `rpm-ostree install`:
https://blogs.gnome.org/alexl/2024/01/15/testing-composefs-in-silverblue/

composefs/fs-verity support is also blocked on the upstream work for UKIs in Fedora Atomic before this can provide meaningful security (otherwise an attacker can swap out the initramfs with one that loads a malicious userspace).

Due to how dm-verity works, it could also not be retrofitted into existing installations, since it requires a very specific partition layout.

**Comment by RoyalOughtness** [comment]
Closing as upstream 

https://fedoraproject.org/wiki/Changes/FsVerityRPM
